utils/datasets_util.py

```diff
@@ -0,0 +1,259 @@
+import numpy as np
+import os
+import os.path as op
+import glob
+import h5py
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+
+USPS_PATH = "/home/dbash_google_com/datasets/usps.h5"
+
+def hdf5(path, data_key="data", target_key="target"):
+  """
+      loads data from hdf5:
+      - hdf5 should have 'train' and 'test' groups
+      - each group should have 'data' and 'target' dataset or spcify the key
+      - flatten means to flatten images N * (C * H * W) as N * D array
+  """
+  with h5py.File(path, "r") as hf:
+    train = hf.get("train")
+    x_tr = train.get(data_key)[:]
+    y_tr = train.get(target_key)[:]
+    test = hf.get("test")
+    x_te = test.get(data_key)[:]
+    y_te = test.get(target_key)[:]
+    x_tr = x_tr.reshape((x_tr.shape[0], 16, 16, 1))*255.
+    x_te = x_te.reshape((x_te.shape[0], 16, 16, 1))*255.
+  return x_tr, y_tr, x_te, y_te
+
+def get_usps_tf_dataset(path):
+  """returns tf.data.Dataset object with USPS dataset.
+
+  Arguments:
+  path: path to h5 file."""
+
+  x_train, y_train, x_test, y_test = hdf5(path)
+  print(x_train.shape, y_train.shape)
+  x_train, y_train = tf.constant(x_train), tf.constant(y_train)
+  x_test, y_test = tf.constant(x_test), tf.constant(y_test)
+
+  train_dataset = tf.data.Dataset.from_tensor_slices((x_train, y_train))
+  test_dataset = tf.data.Dataset.from_tensor_slices((x_test, y_test))
+
+  return train_dataset, test_dataset
+
+
+def get_split_dataset(name, data_dir, split=0.8):
+  """Returns tf.data.Dataset objects for the quickdraw dataset,
+
+  Arguments:
+  name: dataset name.
+  data_dir: the dataset files will be stored in this folder.
+  split: training split."""
+  percent = int(split*100)
+  split_arg = ["train[:%i%%]" % percent, "train[%i%%:]" % percent]
+  datasets, info = tfds.load(name, data_dir=data_dir,
+                             with_info=True,
+                             as_supervised=True,
+                             split=split_arg)
+
+
+  return datasets, info
+
+def get_dataset_from_directory(dir_path, split):
+  """Creates a dataset instance given a dataset directory.
+
+  Arguments:
+  dir_path: path to the dataset files.
+  split: test split."""
+  def get_label(file_path):
+    # convert the path to a list of path components
+    parts = tf.strings.split(file_path, os.path.sep)
```

Comment:
Here and later if there are other occurrences:

```suggestion
    parts = tf.strings.split(file_path, op.sep)
```